app/backend/src/couchers/metrics.py

```diff
@@ -39,35 +39,143 @@ def observe_in_servicer_duration_histogram(method, user_id, status_code, excepti
     servicer_duration_histogram.labels(method, user_id is not None, status_code, exception_type).observe(duration_s)
 
 
-def _get_active_users_5m():
-    with session_scope() as session:
-        return session.execute(
-            select(func.count())
-            .select_from(User)
-            .where(User.is_visible)
-            .where(User.last_active > now() - timedelta(minutes=5))
-        ).scalar_one()
+def _make_gauge_from_query(name, description, statement):
+    """
+    Given a name, description and statement that is a sqlalchemy statement, creates a gauge from it
 
+    statement should be a sqlalchemy SELECT statement that returns a single number
+    """
 
-active_users_5m_gauge = Gauge(
-    "couchers_active_users_5m",
-    "Number of active users in the last 5 min",
-    registry=main_process_registry,
+    def func():
+        with session_scope() as session:
+            return session.execute(statement).scalar_one()
+
+    gauge = Gauge(
+        name,
+        description,
+        registry=main_process_registry,
+    )
+    gauge.set_function(func)
+    return gauge
+
+
+active_users_gauges = [
+    _make_gauge_from_query(
+        f"couchers_active_users_{name}",
+        f"Number of active users in the last {description}",
+        (select(func.count()).select_from(User).where(User.is_visible).where(User.last_active > now() - interval)),
+    )
+    for name, description, interval in [
+        ("5m", "5 min", timedelta(minutes=5)),
+        ("24h", "24 hours", timedelta(hours=24)),
+        ("1month", "1 month", timedelta(days=31)),
+        ("3month", "3 months", timedelta(hours=92)),
+        ("6month", "6 months", timedelta(hours=183)),
+        ("12month", "12 months", timedelta(hours=365)),
+    ]
+]
+
+users_gauge = _make_gauge_from_query(
+    "couchers_users", "Total number of users", select(func.count()).select_from(User).where(User.is_visible)
 )
-active_users_5m_gauge.set_function(_get_active_users_5m)
 
+man_gauge = _make_gauge_from_query(
+    "couchers_users_man",
+    "Total number of users with gender 'Man'",
+    select(func.count()).select_from(User).where(User.is_visible).where(User.gender == "Man"),
+)
 
-def _get_users():
-    with session_scope() as session:
-        return session.execute(select(func.count()).select_from(User).where(User.is_visible)).scalar_one()
+woman_gauge = _make_gauge_from_query(
+    "couchers_users_woman",
+    "Total number of users with gender 'Woman'",
+    select(func.count()).select_from(User).where(User.is_visible).where(User.gender == "Woman"),
+)
```

Comment:
Here an additional one for the rate between genders, or a normalized growth index could be nice

Reply:
You can compute ratios etc in grafana!